src/Integration.Vsix/SonarLintNotificationsPackage.cs

```diff
@@ -46,40 +49,84 @@ public sealed class SonarLintNotificationsPackage : Package
         private const string NotificationDataKey = "NotificationEventData";
 
         private readonly IFormatter formatter = new BinaryFormatter();
-        private readonly NotificationIndicator notificationIcon = new NotificationIndicator();
+        private NotificationIndicator notificationIcon;
 
         private IActiveSolutionBoundTracker activeSolutionBoundTracker;
         private ISonarQubeNotificationService notifications;
-        private ILogger sonarLintOutput;
+        private ILogger logger;
         private NotificationData notificationData;
         private bool disposed;
 
-        protected override void Initialize()
+        protected override System.Threading.Tasks.Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            AddOptionKey(NotificationDataKey);
-            base.Initialize();
+            JoinableTaskFactory.RunAsync(Init);
+            return System.Threading.Tasks.Task.CompletedTask;
+        }
+
+        private async System.Threading.Tasks.Task Init()
+        {
```

Comment:
I've split the initialisation into two parts - the stuff that can be done on the background thread, and the stuff that has to be done on the UI thread. The UI initialisation is in a separate method.